### DISCUSSION TITLE
Subscriptions failing behind Traefik

### DISCUSSION THREAD

**Discussion by ItsRobbAllen** [ravendb/ravendb#14116] [Q&A] [answered]
For local development, I have RavenDb running in a Docker container, using Traefik as a gateway. I can connect to Raven for most everything just fine via `http://ravendb.local` using the following settings
```
      - traefik.enable=true
      - traefik.http.routers.ravendb-rtr.rule=Host(`ravendb.local`)
      - traefik.http.services.ravendb-svc.loadbalancer.server.port=8080
      - traefik.tcp.routers.ravendb.rule=HostSNI(`*`)
      - traefik.tcp.routers.ravendb.entrypoints=ravendb
      - traefik.tcp.routers.ravendb.service=ravendb
      - traefik.tcp.services.ravendb.loadbalancer.server.port=38888
```

However, whenever I try to set up a Subscription, I get the following error

`Failed to connect to http://e6bcca3de357:8080. Please make sure this url is accessible from your browser.`

The URL changes from time to time, which I assume is just the docker container renaming itself. Port 8080 is what Traefik uses for its UI from the outside.

I'm pretty sure there's something simple I'm missing here to get this to work. Anyone have any ideas?

**Comment by arekpalinski** [comment]
Just to make sure, this is the error that you get when trying to create a subscription from the RavenDB Studio right?

@ml054 and @gregolsky can you have a look?

**Comment by gregolsky** [comment, marked as answer]
@ItsRobbAllen did you set up `PublicServerUrl` and `PublicServerUrl.Tcp`?

https://ravendb.net/docs/article-page/5.3/csharp/server/configuration/core-configuration#publicserverurl
https://ravendb.net/docs/article-page/5.3/csharp/server/configuration/core-configuration#publicserverurl.tcp

Can you paste your environment variables/settings.json that your start your container with?
Another setting of interest is `ServerUrl and `ServerUrl.Tcp`. What are these set to?

**Reply by ItsRobbAllen** [reply]
Bingo! None of that is set because my setup is literally a few lines in a docker-compose.yaml file. I need to set up a volume & put a config file in it for Raven to pick up.

For now, is there anyway to change this manually for testing?

**Reply by ayende** [reply]
You can pass that as an env variable, IIRC.
`RavenDB_PublicServerUrl` should work

**Reply by ItsRobbAllen** [reply]
`RAVEN_PublicServerUrl ` was the key and that works perfectly, even easier for what we're doing since this is a dev install.

Thank you all. I knew it was something easy like that.
